timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -166,9 +168,13 @@ def get_freq(series):
         self._cached_freq = freq
         return freq
 
+    @deprecated("Please use `TimeSeriesDataFrame.iter_item_ids()` instead.")
     def iter_items(self) -> Iterable[Any]:
         return iter(self._item_index)
 
+    def iter_item_ids(self) -> Iterable[Any]:
```

Comment:
Should we maybe expose `TimeSeriesDataFrame._item_index` as a public attribute `TimeSeriesDataFrame.item_ids`?
```python
@property
def item_ids(self):
    return self.index.unique(level=ITEMID)
```

There are many places in the code where we're doing something inefficient to get this information (e.g., `list(ts_dataframe.iter_items())` or `ts_dataframe.index.get_level_values(0).unique()`), and I don't really see downsides of exposing this attribute. 
- With the implementation above it will be impossible to mutate the underlying dataframe by messing with the `item_ids`.
- We are not really saving memory by using the iterator, since the array is anyway instantiated inside `iter` (and the size of the array is tiny compared to the size of the entire TimeSeriesDataFrame, so I don't think it's a concern).

We can expose both the iterator and the index with `.item_ids` and `.iter_item_ids()`.

Reply:
I agree 100%.